src/mlpack/methods/ann/layer/resize_impl.hpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file resize_impl.hpp
+ * @author Kris Singh
+ *
+ * Implementation of the Resize layer class also known as fully-connected layer
+ * or affine transformation.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RESIZE_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RESIZE_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include <mlpack/methods/ann/image_functions/bilinear_function.hpp>
+#include "resize.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<class InterpolationType,
+         typename InputDataType,
+         typename OutputDataType
+         >
+Resize<InterpolationType, InputDataType, OutputDataType>::
+Resize(InterpolationType policy): policy(policy)
+{
+  // Nothing to do here.
+}
+
+template<class InterpolationType,
+         typename InputDataType,
+         typename OutputDataType
+         >
+template<typename eT>
+void Resize<InterpolationType, InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>&& input, arma::Mat<eT>&& output)
+{
+  policy.UpSample(input, output);
+}
+
+template<class InterpolationType,
+         typename InputDataType,
+         typename OutputDataType
+         >
+template<typename eT>
+void Resize<InterpolationType, InputDataType, OutputDataType>::Backward(
+    const arma::Mat<eT>&& /* input */, arma::Mat<eT>&& gy, arma::Mat<eT>&& g)
+{
+  policy.DownSample(gy, g);
```

Comment:
Hmmm... I think it is correct only in case of bilinear interpolation. For example quadratic interpolation would require the `input` matrix in order to perform the backward pass (If I am not mistaken). So, I guess we should add that parameter to the policy API.
Moreover, I think it isn't too evident that the backward pass is just the inverse interpolation of the backpropogated error in case of bilinear interpolation. I guess we should add a comment in order to describe that.

Reply:
Fixed the api. But not sure about the comment.